homeassistant/components/reolink/switch.py

```diff
@@ -0,0 +1,106 @@
+"""This component provides support for Reolink switch entities."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from reolink_aio.api import Host
+
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ReolinkData
+from .const import DOMAIN
+from .entity import ReolinkCoordinatorEntity
+
+
+@dataclass
+class ReolinkSwitchEntityDescriptionMixin:
+    """Mixin values for Reolink switch entities."""
+
+    value: Callable[[Host, int | None], bool]
+    method: Callable[[Host, int | None, bool], Any]
+
+
+@dataclass
+class ReolinkSwitchEntityDescription(
+    SwitchEntityDescription, ReolinkSwitchEntityDescriptionMixin
+):
+    """A class that describes switch entities."""
+
+    supported: Callable[[Host, int | None], bool] = lambda api, ch: True
+
+
+SWITCH_ENTITIES = (
+    ReolinkSwitchEntityDescription(
+        key="floodlight",
+        name="Floodlight",
+        icon="mdi:spotlight-beam",
+        supported=lambda api, ch: api.supported(ch, "floodLight"),
+        value=lambda api, ch: api.whiteled_state(ch),
+        method=lambda api, ch, value: api.set_whiteled(ch, state=value),
+    ),
```

Comment:
This is a "light" and should thus be a light entity?

Reply:
These are the build in floodlight LEDs of the camera:
![afbeelding](https://user-images.githubusercontent.com/14811189/218450058-3b720e70-3ae9-4e19-8a3f-1413689b62b6.png)

The control of this floodlight is a bit complicated:
This switch can be used to manually turn the LEDs ON, while this switch is ON they will always be ON.
When this switch is OFF, the floodlight will be controlled based on the internal camera `Floodlight mode` see select entity in this PR: https://github.com/home-assistant/core/pull/87946
`Floodlight mode OFF`: floodlight will always be off.
`Floodlight mode Auto`: floodlight will turn on when an event (person, vehicle, motion etc) is detected by the camera and turn off a couple of seconds after the event seeses.
`Floodlight mode Schedule`: floodlight will be on/off based on the internal camera time schedule.

Then we have a single floodlight brightness number (see PR https://github.com/home-assistant/core/pull/87932), which controls the brightness of the LEDs but this brightness is shared between this switch and the `Floodlight mode`, so if a user does not use this manual switch and leaves it OFF, we still need to be able to access this brightness to control the LED brightness when the camera internally switches the LEDs on (mode `Auto` or `Schedule`).
Therefore I can not incorperate the brightness in the light entity as would normally be done.
I think you have to use the light.turn_on service to change brightness right?

I was a bit confused wether or not to implement this as a light entity or not, and decided for a switch, number and select entity.
@frenck Which entities would you recommond for these 3 features (manual on/off, mode, brightness)?

Reply:
> Which entities would you recommond for these 3 features (manual on/off, mode, brightness)?

Select Entities.

Reply:
Alright, I will change them to light entities

Reply:
I have removed the Floodlight switch in favor of a light entity in this PR: https://github.com/home-assistant/core/pull/88619